rhsm/rhsm-context.c

```diff
@@ -538,8 +542,9 @@ rhsm_context_constructed (GObject *object)
         }
     }
 
-  /* If we have conf coming from /etc/rhsm-host, most probably we need to replace /etc/rhsm */
-  if (g_str_has_prefix (ctx->conf_file, CONFIG_DIR_HOST))
+  /* If we have conf existed and coming from /etc/rhsm-host, most probably we need to replace /etc/rhsm. */
+  if (g_file_test (ctx->conf_file, G_FILE_TEST_EXISTS) && 
```

Comment:
If this works, I think it makes sense to me...

But I find this whole thing *very* confusing :smile: 

I was actually very surprised to discover that `registry.access.redhat.com/ubi9/ubi:latest` contains the full `subscription-manager` package.  That makes it much bigger and seems to deeply conflict with the "subscriptions come from the host" model.

Reply:
Another way to look at this is today we ship `subscription-manager-rhsm-certificates` (but *not* `subscription-manager`) in the rhel-coreos base image.  Can we add a symlink (is that what's needed?) that package instead (or in addition)?

Reply:
For ubi9, there are some changes in [subscription-manager](https://github.com/candlepin/subscription-manager/pull/3171), if detect running in container on OCP, will consider as "not container", so the config file will be default `/etc/rhsm/rhsm.conf`, after load config, the ca cert dir is set correctly under /etc/rhsm. (I have no idea how dnf interact with subscription manager)

For rhel-coreos image(not include subscription-manager), if running in container the config file will be set by default `/etc/rhsm-host/rhsm.conf`(which does not exist), then we get the repo ca cert file is `/etc/rhsm-host/ca/redhat-uep.pem` (this file is existed).
According to [code](https://github.com/rpm-software-management/librhsm/blob/5e0674cf389f14174208641ec411ba7be448d5e3/rhsm/rhsm-context.c#L541), if conf is under `/etc/rhsm-host`, will update ca cet dir from `/etc/rhsm` to `/etc/rhsm-host`, and finally get `/etc/rhsm-host-host/ca/redhat-uep.pem`, the path is not correct and fail.
```
Before replace:
conf=/etc/rhsm-host/rhsm.conf, ca=/etc/rhsm-host/ca, repo=/etc/rhsm-host/ca/redhat-uep.pem
After replace:
conf=/etc/rhsm-host/rhsm.conf, ca=/etc/rhsm-host-host/ca, repo=/etc/rhsm-host-host/ca/redhat-uep.pem
```

If we want rhcos container to behave the same as ubi9, as @colin suggested, maybe include rhsm.conf on rhcos is an option?

Reply:
Read more about this, maybe skip to replace if `ca_cert_dir` already has `/etc/rhsm-host` looks more safe?